docs/enhancements/fep-000-api-device-fleet.md

```diff
@@ -0,0 +1,166 @@
+implement and grow the flightctl service.
+
+### Goals
+
+* Define the format and behavior of the Device and Fleet APIs
+
+* Allow the management of individual fleetless devices on a smaller scale,
+  as an integration mechanism for services like AAP and ACM.
+
+* Allow the management of fleets of devices on a larger scale, via gitops or
+  API.
+
+### Non-Goals
+
+
+
+## Proposal
+
+### Device API
+
+The device API is divided into multiple endpoints, some user facing, and other device/facing,
+this design facilitates cleaner RBAC control of the endpoint access.
+
+#### `/device/{name}` User/Admin facing API
+
+This endpoint can be accessed by the administrator or external systems to
+manage individual devices, also internally from the fleet specifications
+for devices.
+
+When managing a device through a fleet, flightctl is responsible
+of maintaining the device spec in sync with the fleet spec; in this case
+write access is only possible for the modification of labels and other metadata
+through this API.
+
+The proposed changes include:
+
+* Reincorporating device.spec as a visible part through this fronted API.
+* Adding a managementMode field to the device.spec, this field can be set to "fleet" or "manual".
+* The device.spec.os.image contains a reference to the image to be used for the device.
+* The device.spec.containers contains a list of container patterns to be watched on the device (as in the fleet spec)
+* The device.spec.systemd contains a list of systemd service patterns to be watches on the device (as in the fleet spec)
+* The device.spec.config contains a list of config items
+* The config items could be inline definition of config files (inline)
+* Some config items reference exteral configurations (k8s-secret, gitref- ...)
+* External references (image or config items) must support floating tags (latest, main, v1.0),
+  some external items are actually non-versioned, so are inherently floating.
+
+##### Expected behaviour regarding external references with floating tags
+
+Changes on external references is monitored and acted on (git branches,
+device images, secrets), the device controller should re-evaluate the device
+spec and apply the changes.
+
+Several mechanisms can be used to track external references:
+* Polling
+* Webhooks
+
+#### `/device/{name}/rendered` (device side)
+This endpoint is used by the device agent to retrieve the rendered device spec, this endpoint is
+read-only, each device can only access it's own device rendered spec based on the device
+certificate identification or an administrator (for debug purposes).
+
+The proposed changes include:
+* adding a version field, which identifies the version of the configuration.
+* rendering the config details in yaml format instead of a json blob (better for human readability while debugging)
+* the config field is a list of *inline* config items. Using a separate schema for this field which won't allow other config
+  types with external references.
```

Comment:
```suggestion
* the config field is a list of *inline* config items. Using a separate schema for this field which won't allow other config
  types with external references.
```

From the discussion, this would remain as a json blob string instead.